java/code/src/com/suse/manager/webui/services/impl/SystemQuery.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2016 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.services.impl;
```

Comment:
yeah its not the right place i just did not move it yet since i created the file. I would actually even like to move it out of `webui` since there is already so much stuff in there that does not belong there. You have an idea where it would fit because i currently don't see any place that makes direct sense.

Reply:
How would you imagine the package structure to look like eventually, or if we started from scratch? I guess it should reflect the architecture somehow, which is currently more true for the `com.redhat` packages as for the `com.suse` ones. There is the `manager` subpackage, but `SystemQuery` is a *service* interface. What exactly is the difference between a `*Manager` class and a `*Service`? Will we need both services and manager classes in the future?

We won't answer these questions here, but we should start thinking about concrete next steps to move towards a better overall structure. For now what about creating a `services.iface` subpackage and move it there? Then with another PR we could move the whole `services` package out of `webui`. This is just a suggestion, other ideas are welcome of course.

Reply:
i moved it to  `services.iface` for now. i guess the package structure has to be its own topic once we decide to tackle it.